jhack/utils/event_recorder/client.py

```diff
@@ -322,6 +335,8 @@ def _install(unit: str):
         print("Alright.")
 
     print("Shelling over recorder script...")
+    _ensure_recorder_dir(unit)
```

Comment:
in what situations could this path not exist?

Reply:
I was testing a kubernetes core deployment using vms generated in maas, the directory was not there although I did not change anything after juju agent installation on these machines. The version I used was 3-0-3

Reply:
I mean, the recorder dir is the directory where the charm.py file is (`src`). I don't know why you don't have a src folder in the first place, but my point is: without a src folder (and thus without a charm.py file), what are we even doing with the recorder?

I must be missing something essential here... maybe we should have a call to discuss?